sysboundary/ionosphere.cpp

```diff
@@ -1303,19 +1303,31 @@ namespace SBC {
          b[1] = this->dipoleField(r[0],r[1],r[2],Y,0,Y);
          b[2] = this->dipoleField(r[0],r[1],r[2],Z,0,Z);
 
-         std::array<int32_t, 3> fsgridCell = getLocalFsGridCellIndexForCoord(technicalGrid,r);
-         if(technicalGrid.get(fsgridCell[0],fsgridCell[1],fsgridCell[2])->sysBoundaryFlag == sysboundarytype::NOT_SYSBOUNDARY) {
-            const std::array<Real, 3> perB = interpolatePerturbedB(
-               perBGrid,
-               dPerBGrid,
-               technicalGrid,
-               reconstructionCoefficientsCache,
-               fsgridCell[0],fsgridCell[1],fsgridCell[2],
-               r
-            );
-            b[0] += perB[0];
-            b[1] += perB[1];
-            b[2] += perB[2];
+                  std::array<int32_t, 3> fsgridCell = getGlobalFsGridCellIndexForCoord(technicalGrid,r);
+         const std::array<int32_t, 3> localStart = technicalGrid.getLocalStart();
+         const std::array<int32_t, 3> localSize = technicalGrid.getLocalSize();
+         // Make the global index a local one, bypass the fsgrid function that yields (-1,-1,-1) also for ghost cells.
+         fsgridCell[0] -= localStart[0];
+         fsgridCell[1] -= localStart[1];
+         fsgridCell[2] -= localStart[2];
+         
+         if(fsgridCell[0] > localSize[0]+1 || fsgridCell[1] > localSize[1]+1 || fsgridCell[2] > localSize[2]+1
+            || fsgridCell[0] < -1 || fsgridCell[1] < -1 || fsgridCell[2] < -1) {
+            cerr << "Oops!\n";
```

Comment:
A more descriptive error message would be good!

Reply:
Will do. :) *Shouldn't* happen once we've sorted out stuff – but I just got it testing BS in a very coarse setup...

Reply:
These messages have become more descriptive – and rarer, too...